Диплом 2

MargaritaGorshneva2-MD-25 opened the pull request.

MargaritaGorshneva2-MD-25 committed:
Add files via upload

DolgoarshinnykhLG commented on __pycache__/data.cpython-312.pyc:
Нужно исправить: Эта директория лишняя в проекте. Необходимо создать .gitignore в корне проекта и добавить туда объекты не относящиеся к проекту, а затем удалить их из ветки

DolgoarshinnykhLG commented on classes/order.py:
```diff
@@ -0,0 +1,32 @@
+
+class Order:
+
+    @allure.title('Получение хэш ингредиентов')
```
Необходимо исправить здесь и далее: allure.title следует использовать для аннотаций тестов. Для шага теста (или метода вызываемого в нём) лучше подходит allure.step

DolgoarshinnykhLG commented on data.py:
```diff
@@ -0,0 +1,9 @@
+
+
+class Endpoint:
+    CREATE_USER = 'api/auth/register'
+    DEL_AND_CHANGE_USER = 'api/auth/user'
+    LOGIN_USER = 'api/auth/login'
+    CREATE_AND_GET_ORDER = 'api/orders'
+    DATA_ING = 'api/ingredients'
```
Можно лучше: url-адреса можно конкатенировать с базовым (URL), так будет проще поддерживать несколько стендов

DolgoarshinnykhLG commented on tests/test_create_order.py:
```diff
@@ -0,0 +1,55 @@
+    @allure.title('Можно создать заказ с авторизацией')
+    @allure.description('Регистрирую пользователя, авторизуюсь, получаю хэш ингредиентов, создаю заказ')
+    def test_create_order_with_authorization_code_200(self, generating_the_user_and_delete_the_user):
+        data_user = _payload(generating_the_user_and_delete_the_user)
+        req_cr_user = User.create_user(data_user)
+        req_log_user = User.log_user(data_user)
+        req_data_ing = Order.get_data_ing()
```
Можно лучше здесь и далее: шаг с созданием пользователя дублируется в начале нескольких тестов и является предусловием к этим тестам. Можно вынести его в фикстуру

DolgoarshinnykhLG commented on tests/test_create_user.py:
```diff
@@ -0,0 +1,61 @@
+        assert req_re_cr_user.json()['success'] == False
+        assert req_re_cr_user.status_code == HTTPStatus.FORBIDDEN
+
+    @allure.title('Нельзя создать пользователя c пустым полем name')
+    def test_create_user_with_empty_field_name_code_403(self, generating_the_user_and_delete_the_user):
+        data_user = _payload(generating_the_user_and_delete_the_user)
+        data_user['name'] = ''
+        req_cr_user = User.create_user(data_user)
+        assert req_cr_user.json()['success'] == False
+        assert req_cr_user.status_code == HTTPStatus.FORBIDDEN
+
+    @allure.title('Нельзя создать пользователя c пустым полем email')
+    def test_create_user_with_empty_field_email_code_403(self, generating_the_user_and_delete_the_user):
+        data_user = _payload(generating_the_user_and_delete_the_user)
+        data_user['email'] = ''
+        req_cr_user = User.create_user(data_user)
+        assert req_cr_user.json()['success'] == False
+        assert req_cr_user.status_code == HTTPStatus.FORBIDDEN
+
+    @allure.title('Нельзя создать пользователя c пустым полем password')
+    def test_create_user_with_empty_field_password_code_403(self, generating_the_user_and_delete_the_user):
+        data_user = _payload(generating_the_user_and_delete_the_user)
+        data_user['password'] = ''
+        req_cr_user = User.create_user(data_user)
+        assert req_cr_user.json()['success'] == False
+        assert req_cr_user.status_code == HTTPStatus.FORBIDDEN
```
Можно лучше: шаги тестов очень схожи. Можно их объединить с помощью параметризации

MargaritaGorshneva2-MD-25 committed:
Доработка

Доработала всё с пометкой "Нужно исправить" и частично "Можно лучше". Что именно не так с директорией _pycache_ я так и не поняла. .gitignore создан был изначально и при открывании проекта он не показывает этой папки. Если её нужно удалить, то в будущем так и сделаю, но как удалить её из проекта на гитхабе я не разобралась.
